### ISSUE TITLE
Stylint `errormessage` is no longer valid

### ISSUE THREAD

**Issue by emmabukacek** [vim-syntastic/syntastic#1912]
## Bug
* The stylint default reporting was changed in this commit:
  * https://github.com/SimenB/stylint/commit/b041916c84322b8073c0a333fbe2b900ff7331ef#diff-23f2fc2985683e05d92872f3572515b4
* Existing stylint `errorformat` still uses old format.
  * https://github.com/vim-syntastic/syntastic/blob/master/syntax_checkers/stylus/stylint.vim#L24

## Requested Change
* Update `errormessage` in `stylint` `syntax_checker` to match new format.

## Old Output
```
[Warning|Error]: message here
File: filenamehere.styl
Line: 340: white-space pre
```

## New Output
(Results are tabularized)
```
filename.styl
340:11  [warning|error]  message here  stylint_rule_here
```

This is what I've got so far, but I'm not even sure if I'm close:

```viml
let errorformat =
    \ '%A%f%l:%c %s %tarning %m %.%#,' .
    \ '%A%f%l:%c %s %terror %m %.%#,'
```

I'd be happy to submit a PR for this work, but I might need a bit of direction with the file name (and following newline) in addition to properly parsing errors and warnings.

**Comment by lcd047** [comment]
What a terrible idea.  The new error format is ambiguous.  I suppose the proper fix for this would be to write a JSON reporter for `stylint`, and make syntastic use it.  A less proper fix would be to write some kind of `preprocess` function to adjust errors before trying to parse them.

However, I'm still recovering from surgery, and that makes me (1) even more cranky than usual, and (2) unable to type more than a few lines.  So I won't be looking too hard for workarounds.  Something like this will have to do instead: f3a5344.  Sorry about that.

**Comment by ebukacek-rms** [comment]
:(

Can't blame you; thanks for taking a look into this so quickly!

I'll see about opening an issue here soon in the `stylint` repo to see about returning the default reporter either as the default or providing an option for it. I'll report back here if anything changes.

Thanks again!

**Comment by lcd047** [comment]
I committed a fix (1c89d2a), but this is fragile and it's likely to silently drop some messages.  As I said, the proper solution would be to write a JSON reporter for `stylint`.
